src/transmitter/index.ts

```diff
@@ -114,3 +112,15 @@ async function retryRequest(verb: NeedleHttpVerbs, url: string, payload: object)
 
   return {response, attempt};
 }
+
+function shouldRetryRequest(err: IRequestError, networkErrorMessages: string[], stillHaveRetries: boolean): boolean {
```

Comment:
I think that `networkErrorMessages` is really only used within this `shouldRetryRequest()` function so it probably doesn't need to be passed as a variable any more. I think also currently the array gets allocated every time the function `retryRequest()` is called, so we probably don't need to do that! How about moving the `networkErrorMessages[]` as a const global variable in the current module?